### PR TITLE
[LRE] Add LRE space

### PR DESCRIPTION
**Pull request by kevin-verschaeve**
"La république ensemble"

[ch11802]

### CODE REVIEW

File: templates/adherent/_list_my_sections.html.twig [hunk 1]
```diff
@@ -102,3 +102,10 @@
         </a>
     </li>
 {% endif %}
+{% if app.user.isLre %}
```
Comment (renanbr) [inline templates/adherent/_list_my_sections.html.twig:105]: Should we check for `ROLE_LRE` instead?

Comment (kevin-verschaeve) [reply templates/adherent/_list_my_sections.html.twig:105]: I did the same as the other checks, anyway, in this case, this is the same, as ROLE_LRE is present when isLre returns true :)

File: src/Collection/AdherentCharterCollection.php [hunk 2]
```diff
@@ -48,6 +49,13 @@ public function hasSenatorialCandidateCharterAccepted(): bool
         });
     }
 
+    public function hasLRECharterAccepter()
```
Comment (ottaviano) [inline src/Collection/AdherentCharterCollection.php:52]: typehint please

File: src/Collection/AdherentCharterCollection.php [hunk 3]
```diff
@@ -65,6 +73,9 @@ public function hasCharterAcceptedForType(string $type): bool
 
             case AdherentCharterTypeEnum::TYPE_SENATORIAL_CANDIDATE:
                 return $this->hasSenatorialCandidateCharterAccepted();
+
+            case AdherentCharterTypeEnum::TYPE_LRE:
+                return $this->hasLRECharterAccepter();
```
Comment (ottaviano) [inline src/Collection/AdherentCharterCollection.php:78]: `hasLRECharterAccepted`

File: src/Entity/Adherent.php [hunk 4]
```diff
@@ -456,6 +456,14 @@ class Adherent implements UserInterface, UserEntityInterface, GeoPointInterface,
      */
     private $senatorialCandidateManagedArea;
 
+    /**
+     * @var LREArea|null
+     *
+     * @Assert\Valid
```
Comment (ottaviano) [inline src/Entity/Adherent.php:462]: It seems that `LREArea` entity does not have any `@Assert`, is it an oversight?

File: src/Entity/LREArea.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     */
+    private $id;
+
+    /**
+     * @var ReferentTag|null
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\ReferentTag")
```
Comment (ottaviano) [inline src/Entity/LREArea.php:27]: do we really need to keep it nullable? if not, I propose to do: 
- `@JoinColumn(nullable=false)`
- `__construct(ReferentTag $tag)`

File: src/Entity/LreArea.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        $this->departmentTag = $tag;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getDepartmentTag(): ?ReferentTag
```
Comment (ottaviano) [inline src/Entity/LreArea.php:42]: `?ReferentTag` => `ReferentTag`

File: src/Entity/LreArea.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        return $this->departmentTag;
+    }
+
+    public function setDepartmentTag(?ReferentTag $departmentTag): void
```
Comment (ottaviano) [inline src/Entity/LreArea.php:47]: `?ReferentTag` => `ReferentTag`

**Pull request merged**